File: check.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
)

func main() {
	got, err := ioutil.ReadFile("got.txt")
	if err != nil {
		panic(err)
	}

	expected, err := ioutil.ReadFile("expected.txt")
	if err != nil {
		panic(err)
	}

	got_lines := strings.Split(string(got), "\n")
	expected_lines := strings.Split(string(expected), "\n")
	failed := false

	for i := 0; i < len(got_lines)-1; i++ {
		if got_lines[i] == "#" {
			continue
		}
		if expected_lines[i] != got_lines[i] {
			fmt.Printf("Failed at line: %d\n", i)
			fmt.Println("Expected:")
			if i != 0 {
				fmt.Println(expected_lines[i-1])
				fmt.Println(expected_lines[i])
				fmt.Println("\nGot:")
				fmt.Println(got_lines[i-1])
				fmt.Println(got_lines[i])
			} else {
				fmt.Println(expected_lines[i])
				fmt.Println("\nGot:")
				fmt.Println(got_lines[i])
			}
			failed = true
			break
		}
	}
	if !failed {
		fmt.Println("Passed")
	}
}

File: check.js
const fs = require("fs");
const { argv } = require("process");

const startAt = Number(argv[2]);

const got = fs.readFileSync("got.txt").toString().split("\n");
const expected = fs.readFileSync("expected.txt").toString().split("\n");
var fail = false;

for (var i = 0; i < got.length - 1; i++) {
  if (got[i] == "#") continue;
  if (got[i] !== expected[i + startAt]) {
    console.log("Failed at line: ", i + startAt);
    console.log(
      "Expected: \n",
      expected[i + startAt - 1],
      "\n",
      expected[i + startAt]
    );
    console.log("\n\nGot: \n", got[i - 1], "\n", got[i]);
    fail = true;
    break;
  }
}

if (!fail) {
  console.log("Success");
}
